subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/AvailableToolChains.java

```diff
@@ -167,24 +170,25 @@ static private ToolChainCandidate findCygwin() {
         return new UnavailableToolChain("gcc cygwin");
     }
 
-    static private ToolChainCandidate findGcc() {
+    static private Set<ToolChainCandidate> findGccs() {
         GccVersionDeterminer versionDeterminer = GccVersionDeterminer.forGcc(TestFiles.execActionFactory());
 
         List<File> gppCandidates = OperatingSystem.current().findAllInPath("g++");
+        Set<ToolChainCandidate> toolChains = Sets.newLinkedHashSet();
         for (int i = 0; i < gppCandidates.size(); i++) {
             File candidate = gppCandidates.get(i);
             GccVersionResult version = versionDeterminer.getGccMetaData(candidate, Collections.<String>emptyList());
             if (version.isAvailable()) {
-                InstalledGcc gcc = new InstalledGcc("gcc");
+                InstalledGcc gcc = new InstalledGcc("gcc" + " " + version.getVersion());
                 if (i > 0) {
                     // Not the first g++ in the path, needs the path variable updated
                     gcc.inPath(candidate.getParentFile());
                 }
-                return gcc;
+                toolChains.add(gcc);
             }
         }
 
-        return new UnavailableToolChain("gcc");
+        return toolChains;
```

Comment:
Should we still add an `UnavailableToolChain("gcc")` if no GCC is found?

Reply:
Good point. I re-added the unavailable toolchain.